frontend/components/frontpage-feed.js

```diff
@@ -0,0 +1,214 @@
+import Tonic from '@socketsupply/tonic/index.esm.js'
+import { publicKernel as pub } from '../api.js'
+import { gate, combine, mute } from 'piconuro'
+import { isRantID, btok } from '../../blockend/kernel.js'
+import { processText, THEMES } from '../../blockend/picocard.js'
+import Modem56 from 'picostack/modem56.js'
+import { nEl } from '../surgeon.js'
+import { createAlert } from './alert.js'
+
+const TOPIC = 'GLOBAL_RANT_WARNING'
+
+/**
+ * frontpage is working as intended!
+ * TODO: clean up and remove all unused stuff. also remove discover-page since we dont need it anymore.
+ *
+ */
+Tonic.add(class FrontpageFeed extends Tonic {
+  constructor () {
+    super()
+    this.modem = new Modem56()
+    this.isSwarming = true
+
+    this.modem.join(TOPIC, (...a) => pub.spawnWire(...a))
+    console.log('second kernel booted, topic joined', TOPIC)
+    this.reRender(p => p)
+    // this.attachShadow({ mode: 'open' }) // <-- Shadow DOM
+  }
+
+  connected () {
+    this.ready = this.dataset.ready === 'true'
+
+    // Filter rants that exist in local notebook.
+    const $rants = pub.$rants()
+    const $state = gate(
+      combine({
+        rants: $rants,
+        peers: mute(pub.$connections(), c => c.length)
+      })
+    )
+    this.unsub = $state(state => {
+      this.reRender(prev => ({ ...prev, ...state }))
+    })
+    const getKey = async () => {
+      const key = await pub.$key()
+      this.key = btok(key)
+      this.reRender()
+    }
+    getKey()
+  }
+
+  disconnected () { this.unsub() }
+
+  change (ev) {
+    if (Tonic.match(ev.target, '#toggle-swarm')) {
+      this.toggleSwarm(ev.target.checked)
+    }
+  }
+
+  async click (ev) {
+    const el = Tonic.match(ev.target, 'rant[data-id]')
+    if (el) {
+      // const id = Buffer.from(el.dataset.id, 'base64')
+    }
+  }
+
+  render () {
+    const unSortedRants = this.props.rants || []
+    const peers = this.props.peers || 0
+    this.classList.add('pad')
+
+    const listRants = () => {
+      if (!unSortedRants.length) {
+        return this.html`
+          <h5 aria-busy="true">Searching for peers...</h5>
+        `
+      }
+      const rants = unSortedRants.sort((a, b) => b.expiresAt - a.expiresAt) // sort the rants. Display long lived at top.
+      return rants.map(rant => {
+        return this.html`
+          <rant-card rant=${rant} key=${this.key}></rant-card>
+        `
+      })
+    }
+
+    return this.html`
+    <div style="text-align: center;">
+      <h1 style="font-size: 6rem; margin-bottom:0.25rem;">1k</h1>
+      <p>Tin Foil Twitter</p>
```

Comment:
mixed like. don't like using other brands but easy to relate. How about _Real Times_ news network? 😆

Reply:
This was only memes, you pick the slogan :smile_cat: